test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6040,10 +6040,11 @@ def test_release_nodes_excl_server_restart_immed(self):
         self.server.start()
         self.assertTrue(self.server.isUp())
 
-        # make sure job is now running with assigned resources
+        # make sure job is now running after server restart
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        # make sure job is running with assigned resources
         # from the original request
-        self.server.expect(JOB, {'job_state': 'R',
-                                 'Resource_List.mem': '5gb',
+        self.server.expect(JOB, {'Resource_List.mem': '5gb',
```

Comment:
I'm a little confused why this change is required.  I thought that expect() will trigger scheduling cycles unless you tell it not to with saying trigger_sched_cycle=False.

Reply:
Hi @bhroam , The cycle will be triggered if the data does not match, But here the data for exec_host is not available as the job is in Q state. So the test fails.

Reply:
That sounds like a bug in expect() to me.  If I understand what you are saying is that if exec_host was set but didn't match, it'd trigger a cycle.  Since it isn't set, it just fails.  Could you file a bug against expect?  I'll sign off on your workaround here.